internal/controller/ols_app_server_deployment.go

```diff
@@ -0,0 +1,189 @@
+					Name: OLSConfigCmName,
+				},
+			},
+		},
+	}
+	volumes = append(volumes, volume)
+
+	// mount the volumes of api keys secrets and OLS config map to the container
+	volumeMounts := []corev1.VolumeMount{}
+	for secretName, mountPath := range secretMounts {
+		volumeMount := corev1.VolumeMount{
+			Name:      "secret-" + secretName,
+			MountPath: mountPath,
+			ReadOnly:  true,
+		}
+		volumeMounts = append(volumeMounts, volumeMount)
+	}
+	volumeMount := corev1.VolumeMount{
+		Name:      OLSConfigVolumeName,
+		MountPath: OLSConfigMountPath,
+		ReadOnly:  true,
+	}
+	volumeMounts = append(volumeMounts, volumeMount)
+
+	replicas := getOLSServerReplicas(cr)
+	resources := getOLSServerResources(cr)
+
+	deployment := appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      OLSAppServerDeploymentName,
+			Namespace: cr.Namespace,
+			Labels:    DeploymentSelectorLabels,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: replicas,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: DeploymentSelectorLabels,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: DeploymentSelectorLabels,
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:            "lightspeed-service-api",
+							Image:           r.Options.LightspeedServiceImage,
+							ImagePullPolicy: corev1.PullIfNotPresent,
+							Ports: []corev1.ContainerPort{
+								{
+									ContainerPort: OLSAppServerContainerPort,
+									Name:          "http",
+									Protocol:      corev1.ProtocolTCP,
+								},
+							},
+							SecurityContext: &corev1.SecurityContext{
+								AllowPrivilegeEscalation: &[]bool{false}[0],
+							},
+							VolumeMounts: volumeMounts,
+							Env: []corev1.EnvVar{
+								{
+									Name:  "OLS_CONFIG_FILE",
+									Value: path.Join(OLSConfigMountPath, OLSConfigFilename),
+								},
+							},
+							Resources: *resources,
+						},
+					},
+					Volumes:            volumes,
+					ServiceAccountName: OLSAppServerServiceAccountName,
+				},
+			},
+		},
+	}
+
+	if err := controllerutil.SetControllerReference(cr, &deployment, r.Scheme); err != nil {
+		return nil, err
+	}
+
+	return &deployment, nil
+}
+
+// updateOLSDeployment updates the deployment based on CustomResource configuration.
+func (r *OLSConfigReconciler) updateOLSDeployment(ctx context.Context, existingDeployment, desiredDeployment *appsv1.Deployment) error {
+	changed := false
+
+	// Validate deployment annotations.
+	if existingDeployment.Annotations == nil ||
+		existingDeployment.Annotations[OLSConfigHashKey] != r.stateCache[OLSConfigHashStateCacheKey] {
+		updateDeploymentAnnotations(existingDeployment, map[string]string{
+			OLSConfigHashKey: r.stateCache[OLSConfigHashStateCacheKey],
+		})
+		changed = true
+	}
+
+	// Validate deployment replicas.
+	if setDeploymentReplicas(existingDeployment, *desiredDeployment.Spec.Replicas) {
```

Comment:
can we combine these 2 if into 1 using OR please? looks like a growing ladder IMHO.